python/pyiceberg/expressions/visitors.py

```diff
@@ -482,6 +502,12 @@ def visit_less_than(self, term: BoundTerm[L], literal: Literal[L]) -> bool:
     def visit_less_than_or_equal(self, term: BoundTerm[L], literal: Literal[L]) -> bool:
         return term.eval(self.struct) <= literal.value
 
+    def visit_starts_with(self, term: BoundTerm[L], literal: Literal[L]) -> bool:
+        return str(term.eval(self.struct)).startswith(str(literal.value))
```

Comment:
This looks good to me. Can you make sure that we port the tests from Java also to Python: https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/expressions/TestEvaluator.java#L289-L321

This is a very important part of PyIceberg, and we need to make sure that this is tested to avoid correctness issues.